service/controller/resource/alerting/alertmanager/resource.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	labels := make(map[string]string)
+	for k, v := range key.AlertmanagerLabels(cluster) {
+		labels[k] = v
+	}
+
+	var storage monitoringv1.StorageSpec
+	storageSize := resource.MustParse(config.StorageSize)
+	if config.CreatePVC {
+		storage = monitoringv1.StorageSpec{
+			VolumeClaimTemplate: monitoringv1.EmbeddedPersistentVolumeClaim{
+				Spec: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: storageSize,
+						},
+					},
+				},
+			},
+		}
+	} else {
+		storage = monitoringv1.StorageSpec{
+			EmptyDir: &corev1.EmptyDirVolumeSource{
+				SizeLimit: &storageSize,
+			},
+		}
+	}
+
+	address, err := url.Parse(config.Address)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Alertmanager default image runs using the nobody user (65534)
+	var uid int64 = 65534
+	var fsGroup int64 = 65534
+	var runAsNonRoot bool = true
+	var gid int64 = 65534
+	alertmanager := &monitoringv1.Alertmanager{
+		ObjectMeta: objectMeta,
+		Spec: monitoringv1.AlertmanagerSpec{
+			PodMetadata: &monitoringv1.EmbeddedObjectMetadata{
+				Labels: labels,
+			},
+			Version:     config.Version,
+			LogLevel:    config.LogLevel,
+			ExternalURL: address.String(),
+			Replicas:    &replicas,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					// cpu: 100m
+					corev1.ResourceCPU: *key.AlertmanagerDefaultCPU(),
+					// memory: 200m
+					corev1.ResourceMemory: *key.AlertmanagerDefaultMemory(),
+				},
+				Limits: corev1.ResourceList{
+					// cpu: 100m
+					corev1.ResourceCPU: *key.AlertmanagerDefaultCPU(),
+					// memory: 200m
+					corev1.ResourceMemory: *key.AlertmanagerDefaultMemory(),
+				},
+			},
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsUser:    &uid,
+				RunAsGroup:   &gid,
+				RunAsNonRoot: &runAsNonRoot,
+				FSGroup:      &fsGroup,
+			},
+			Storage: &storage,
+			Affinity: &corev1.Affinity{
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+						NodeSelectorTerms: []corev1.NodeSelectorTerm{
+							{
+								MatchExpressions: []corev1.NodeSelectorRequirement{
+									{
+										Key:      "role",
+										Operator: corev1.NodeSelectorOpNotIn,
+										Values: []string{
+											"master",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
+				{
+					MaxSkew:           1,
+					TopologyKey:       "kubernetes.io/hostname",
+					WhenUnsatisfiable: corev1.ScheduleAnyway,
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app.kubernetes.io/name": "alertmanager",
+						},
+					},
+				},
+			},
+			PriorityClassName: "prometheus",
```

Comment:
This name feels super weird now. We should change it to something like `monitoring` but you can leave this for now.

Reply:
I agree and we should do it also for the namespace and so on but this will require a lot of things that I would rather not have in this PR